### PR TITLE
DB encapsulation and interface

### LINKED ISSUE #5: Refactor db logic
We should use encapsulate db structs in a generic Db struct instead of using the psql struct directly.

### PR DESCRIPTION
**Pull request by jonandernovella**
Added struct to wrap db instances and provide interfaces for the new struct.

This PR fixes #5

### CODE REVIEW

**Comment by jonandernovella** [comment]
> This doesn't actually change anything.
> You will have to go back to the drawing board if you intend to be able to do what I think you want.

Correct. I did some changes now which make use of the ifces.

File: cmd/finalize/finalize.go [hunk 1]
```diff
@@ -23,14 +23,14 @@ type Message struct {
 func main() {
 	config := NewConfig()
 	mq := broker.New(config.Broker)
-	db, err := postgres.NewDB(config.Postgres)
+	dbs, err := postgres.NewDB(config.Postgres)
```
Comment (jbygdell) [inline cmd/finalize/finalize.go:26]: Unnecessary renaming
```suggestion
	db, err := postgres.NewDB(config.Postgres)
```

File: cmd/finalize/finalize.go [hunk 2]
```diff
@@ -23,14 +23,14 @@ type Message struct {
...
 	if err != nil {
 		log.Println("err:", err)
 	}
 
+	defer dbs.Close()
```
Comment (jbygdell) [inline cmd/finalize/finalize.go:31]: Unnecessary renaming
```suggestion
	defer db.Close()
```

File: cmd/finalize/finalize.go [hunk 3]
```diff
@@ -47,7 +47,7 @@ func main() {
 			}
 
 			if err == nil {
-				err := postgres.MarkReady(db, message.StableID, message.User, message.FilePath, message.Checksums[0].Value)
+				err := dbs.MarkReady(message.StableID, message.User, message.FilePath, message.Checksums[0].Value)
```
Comment (jbygdell) [inline cmd/finalize/finalize.go:50]:
```suggestion
				err := db.MarkReady(message.StableID, message.User, message.FilePath, message.Checksums[0].Value)
```

File: cmd/ingest/ingest.go [hunk 4]
```diff
@@ -27,14 +27,14 @@ type Message struct {
 func main() {
 	config := NewConfig()
 	mq := broker.New(config.Broker)
-	db, err := postgres.NewDB(config.Postgres)
+	dbs, err := postgres.NewDB(config.Postgres)
```
Comment (jbygdell) [inline cmd/ingest/ingest.go:30]: Unnecessary renaming
```suggestion
	db, err := postgres.NewDB(config.Postgres)
```

File: cmd/ingest/ingest.go [hunk 5]
```diff
@@ -27,14 +27,14 @@ type Message struct {
...
 	if err != nil {
 		log.Println("err:", err)
 	}
 
+	defer dbs.Close()
```
Comment (jbygdell) [inline cmd/ingest/ingest.go:35]: Unnecessary renaming
```suggestion
	defer db.Close()
```

File: cmd/verify/verify.go [hunk 6]
```diff
@@ -44,14 +44,14 @@ type Checksums struct {
 func main() {
 	config := NewConfig()
 	mq := broker.New(config.Broker)
-	db, err := postgres.NewDB(config.Postgres)
+	dbs, err := postgres.NewDB(config.Postgres)
```
Comment (jbygdell) [inline cmd/verify/verify.go:47]: Unnecessary renaming
```suggestion
	db, err := postgres.NewDB(config.Postgres)
```

File: cmd/verify/verify.go [hunk 7]
```diff
@@ -44,14 +44,14 @@ type Checksums struct {
...
 	if err != nil {
 		log.Println("err:", err)
 	}
 
+	defer dbs.Close()
```
Comment (jbygdell) [inline cmd/verify/verify.go:52]: Unnecessary renaming
```suggestion
	defer db.Close()
```

File: cmd/verify/verify.go [hunk 8]
```diff
@@ -77,7 +77,7 @@ func main() {
 				continue
 			}
 
-			header, err := postgres.GetHeader(db, message.FileID)
+			header, err := dbs.GetHeader(message.FileID)
```
Comment (jbygdell) [inline cmd/verify/verify.go:80]:
```suggestion
			header, err := db.GetHeader(message.FileID)
```

File: cmd/verify/verify.go [hunk 9]
```diff
@@ -122,9 +122,9 @@ func main() {
 			}
 			key = [32]byte{}
 
-			if ! *message.ReVerify {
+			if !*message.ReVerify {
 				// Mark file as "COMPLETED"
-				if e := postgres.MarkCompleted(db, fmt.Sprintf("%x", hash.Sum(nil)), message.FileID); e != nil {
+				if e := dbs.MarkCompleted(fmt.Sprintf("%x", hash.Sum(nil)), message.FileID); e != nil {
```
Comment (jbygdell) [inline cmd/verify/verify.go:127]:
```suggestion
				if e := db.MarkCompleted(fmt.Sprintf("%x", hash.Sum(nil)), message.FileID); e != nil {
```

File: internal/postgres/postgres.go [hunk 10]
```diff
@@ -41,8 +54,9 @@ func NewDB(c Pgconf) (*sql.DB, error) {
 		log.Errorf("Couldn't ping postgres database (%s)", err)
 		panic(err)
 	}
+	dbs := SQLdb{Db: db}
 
-	return db, err
+	return &dbs, err
```
Comment (jbygdell) [inline internal/postgres/postgres.go:59]:
```suggestion
	return &SQLdb{Db: db}, err
```

File: internal/postgres/postgres.go [hunk 11]
```diff
@@ -41,8 +54,9 @@ func NewDB(c Pgconf) (*sql.DB, error) {
 		log.Errorf("Couldn't ping postgres database (%s)", err)
 		panic(err)
 	}
+	dbs := SQLdb{Db: db}
```
Comment (jbygdell) [inline internal/postgres/postgres.go:57]:
```suggestion

```

File: cmd/finalize/finalize.go [hunk 12]
```diff
@@ -28,9 +28,9 @@ func main() {
 		log.Println("err:", err)
 	}
 
+	defer db.Close()
 	defer mq.Channel.Close()
 	defer mq.Connection.Close()
-	defer db.Close()
```
Comment (jbygdell) [inline cmd/finalize/finalize.go:33]: No point in moving the `db.lose()` above the mq lines

File: cmd/ingest/ingest.go [hunk 13]
```diff
@@ -32,9 +32,9 @@ func main() {
 		log.Println("err:", err)
 	}
 
+	defer db.Close()
 	defer mq.Channel.Close()
 	defer mq.Connection.Close()
-	defer db.Close()
```
Comment (jbygdell) [inline cmd/ingest/ingest.go:38]: same here

File: cmd/verify/verify.go [hunk 14]
```diff
@@ -49,9 +49,9 @@ func main() {
 		log.Println("err:", err)
 	}
 
+	defer db.Close()
 	defer mq.Channel.Close()
 	defer mq.Connection.Close()
-	defer db.Close()
```
Comment (jbygdell) [inline cmd/verify/verify.go:54]: same again

File: cmd/verify/verify.go [hunk 15]
```diff
@@ -26,7 +26,7 @@ type Message struct {
 	FileID       int    `json:"file_id"`
 	ArchivePath  string `json:"archive_path"`
 	FileChecksum string `json:"file_checksum"`
-	ReVerify     *bool   `json: "re_verify"`
+	ReVerify     *bool  `json: "re_verify"`
```
Comment (jbygdell) [inline cmd/verify/verify.go:29]:
```suggestion
	ReVerify     *bool  `json:"re_verify"`
```
This should make the tests pass.

**Pull request merged**